crates/sui-framework/packages/sui-framework/sources/crypto/nitro_attestation.move

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::nitro_attestation;
+
+use sui::clock::{Self, Clock};
+
+public struct NitroAttestationDocument has store, copy, drop  {
+    module_id: vector<u8>,
+    timestamp: u64,
+    digest: vector<u8>,
+    pcrs: vector<vector<u8>>,
+    public_key: Option<vector<u8>>,
+    user_data: Option<vector<u8>>,
+    nonce: Option<vector<u8>>,
+}
+
+/// Internal native function
+native fun verify_nitro_attestation_internal(
+    attestation: &vector<u8>,
+    current_timestamp: u64
+): NitroAttestationDocument;
+
+/// @param attestation: attesttaion documents bytes data. 
+/// @param clock: the clock object.
+///
+/// Returns parsed NitroAttestationDocument after verifying the attestation.
```

Comment:
also document the errors in case of expired cert, etc

Reply:
all verify error is currently casted to a generic VerifyError. any reason this should be more specific?

Reply:
it's ok, but let's have that error defined in move as well and document it

Reply:
i added them here. do you mean anywhere else?
```
#[allow(unused_const)]
/// Error that the feature is not available on this network.
const ENotSupportedError: u64 = 0;
#[allow(unused_const)]
/// Error that the attestation input failed to be parsed. 
const EParseError: u64 = 1;
#[allow(unused_const)]
/// Error that the attestation failed to be verified. 
const EVerifyError: u64 = 2;
#[allow(unused_const)]
/// Error that the pcrs length is invalid. 
const EInvalidPcrLength: u64 = 3;
```

Reply:
oh, i see that you explained each of those errors there, great! so let's just add here something like short like `May abort with the error codes described above`